zsh-history-substring-search.zsh

```diff
@@ -246,7 +246,13 @@ _history-substring-search-begin() {
     # Escape and join query parts with wildcard character '*' as seperator
     # `(j:CHAR:)` join array to string with CHAR as seperator
     #
-    local search_pattern="*${(j:*:)_history_substring_search_query_parts[@]//(#m)[\][()|\\*?#<>~^]/\\$MATCH}*"
+    local search_pattern="${(j:*:)_history_substring_search_query_parts[@]//(#m)[\][()|\\*?#<>~^]/\\$MATCH}*"
+    #
+    # Support anchoring history search to the beginning of the command
+    # 
+    if [[ -z $HISTORY_SUBSTRING_SEARCH_ANCHORED ]]; then
```

Comment:
I suggest adding this variable to the "declare global configuration variables" section near the top of the file